ui/send_page.go

```diff
@@ -560,44 +649,127 @@ func (pg *SendPage) validateAmount(ignoreEmpty bool) bool {
 }
 
 func (pg *SendPage) calculateValues() {
-	pg.transactionFeeValueLabel.Text = "0 DCR"
-	pg.totalCostValueLabel.Text = "0 DCR"
+	defaultActiveValues := fmt.Sprintf("- %s", pg.activeExchange)
+	defaultInactiveValues := fmt.Sprintf("(- %s)", pg.inactiveExchange)
+	noExchangeText := "Exchange rate not fetched"
+
+	pg.activeTransactionFeeValue = defaultActiveValues
+	pg.activeTotalCostValue = defaultActiveValues
+	pg.inactiveTransactionFeeValue = defaultInactiveValues
+	pg.inactiveTotalCostValue = defaultInactiveValues
+
 	pg.calculateErrorText = ""
+	pg.activeTotalAmount = defaultActiveValues
+	pg.inactiveTotalAmount = fmt.Sprintf("- %s", pg.inactiveExchange)
+
+	// default values when exchange is not available
+	if pg.LastTradeRate == "" {
+		pg.activeTransactionFeeValue = defaultActiveValues
+		pg.activeTotalCostValue = defaultActiveValues
+		pg.inactiveTransactionFeeValue = ""
+		pg.inactiveTotalCostValue = ""
+		pg.activeTotalAmount = defaultActiveValues
+		pg.inactiveTotalAmount = noExchangeText
+	}
 
 	if pg.txAuthor == nil || !pg.validate(true) {
 		return
 	}
 
-	amountDCR, _ := strconv.ParseFloat(pg.sendAmountEditor.Text(), 64)
-	amount, err := dcrutil.NewAmount(amountDCR)
+	inputValue, _ := strconv.ParseFloat(pg.sendAmountEditor.Text(), 64)
+
+	var usdExchangeRate, amountUSDtoDCR float64
+	if pg.LastTradeRate != "" {
+		usdExchangeRate, _ = strconv.ParseFloat(pg.LastTradeRate, 64)
+		amountUSDtoDCR = inputValue / usdExchangeRate
+	}
+
+	var amountAtoms int64
+	if pg.activeExchange == "USD" && pg.LastTradeRate != "" {
+		amountAtoms = pg.setDestinationAddr(amountUSDtoDCR)
+		if amountAtoms == 0 {
+			return
+		}
+	} else {
+		amountAtoms = pg.setDestinationAddr(inputValue)
+		if amountAtoms == 0 {
+			return
+		}
+	}
+
+	txFee := pg.getTxFee()
+	if txFee == 0 {
+		return
+	}
+
+	totalCostDCR := txFee + amountAtoms
+
+	switch {
+	case pg.activeExchange == "DCR" && pg.LastTradeRate != "":
+		// calculate total tx amount in USD
+		totalAmountUSD := inputValue * usdExchangeRate
+		txFeeValueUSD := dcrutil.Amount(txFee).ToCoin() * usdExchangeRate
+
+		totalAmountUSDTostring := fmt.Sprintf("%s USD", strconv.FormatFloat(totalAmountUSD, 'f', 7, 64))
+
+		pg.activeTotalAmount = dcrutil.Amount(amountAtoms).String()
+		pg.inactiveTotalAmount = totalAmountUSDTostring
+		pg.activeTransactionFeeValue = dcrutil.Amount(txFee).String()
+		pg.inactiveTransactionFeeValue = fmt.Sprintf("(%f USD)", txFeeValueUSD)
+		pg.activeTotalCostValue = dcrutil.Amount(totalCostDCR).String()
+		pg.inactiveTotalCostValue = fmt.Sprintf("(%s USD)", strconv.FormatFloat(totalAmountUSD+txFeeValueUSD, 'f', 7, 64))
```

Comment:
This looks repeated. To make it cleaner, you can create a data type for it. For each case, assign the respective data properties their respective methods and pass that to a single method that initiates those values to the SendPage properties.
 Also, all that would be cleaner if they were in a separate method.